File: MyInsight/MyInsight/UIFunction/Tabbar/TabBarVC.m
//
//  TabBarVC.m
//  MyInsight
//
//  Created by SongMenglong on 2017/12/1.
//  Copyright © 2017年 SongMenglong. All rights reserved.
//

#import "TabBarVC.h"

#import "BasicVC.h"
#import "AdvanceVC.h"
#import "SeniorVC.h"
#import "OtherVC.h"

@interface TabBarVC ()

@end

@implementation TabBarVC

- (void)viewDidLoad {
    [super viewDidLoad];
    // Demo地址
    // https://github.com/NJHu/iOSProject
    
    self.view.backgroundColor = [UIColor whiteColor];
    // 基础
    BasicVC *basicVC = [[BasicVC alloc] init];
    UINavigationController *basicNavi = [[UINavigationController alloc] initWithRootViewController:basicVC];
    // 进阶
    AdvanceVC *advanceVC = [[AdvanceVC alloc] init];
    UINavigationController *advanceNavi = [[UINavigationController alloc] initWithRootViewController:advanceVC];
    // 高级
    SeniorVC *seniorVC = [[SeniorVC alloc] init];
    UINavigationController *seniorNavi = [[UINavigationController alloc] initWithRootViewController:seniorVC];
    // 其他
    OtherVC *otherVC = [[OtherVC alloc] init];
    UINavigationController *otherNavi = [[UINavigationController alloc] initWithRootViewController:otherVC];
    
    // 设置各个tabbar
    // 基础
    basicVC.tabBarItem.image = [UIImage imageNamed:@"home_nor"];
    basicVC.tabBarItem.selectedImage = [UIImage imageNamed:@"home_sel"];
    basicVC.tabBarItem.title = @"基础";
    basicVC.title = basicVC.tabBarItem.title;
    // 进阶
    advanceVC.tabBarItem.image = [UIImage imageNamed:@"mark_nor"];
    advanceVC.tabBarItem.selectedImage = [UIImage imageNamed:@"mark_sel"];
    advanceVC.tabBarItem.title = @"进阶";
    advanceVC.title = advanceVC.tabBarItem.title;
    // 高级
    seniorVC.tabBarItem.image = [UIImage imageNamed:@"mine_nor"];
    seniorVC.tabBarItem.selectedImage = [UIImage imageNamed:@"mine_sel"];
    seniorVC.tabBarItem.title = @"高级";
    seniorVC.title = seniorVC.tabBarItem.title;
    // 其他
    otherVC.tabBarItem.image = [UIImage imageNamed:@"setting_nor"];
    otherVC.tabBarItem.selectedImage = [UIImage imageNamed:@"setting_sel"];
    otherVC.tabBarItem.title = @"其他";
    otherVC.title = otherVC.tabBarItem.title;
    
    // 设置徽标
    [basicVC.navigationController.tabBarItem setBadgeValue:@"2"];
    
    // 设置tabbar
    [self setViewControllers:@[basicNavi, advanceNavi, seniorNavi, otherNavi] animated:NO];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: MyInsight_Swift/MyInsight_Swift/UIFunction/Reveal/TabBarVC.swift
//
//  TabBarVC.swift
//  MyInsight_Swift
//
//  Created by SongMengLong on 2018/7/22.
//  Copyright © 2018年 SongMengLong. All rights reserved.
//

import UIKit

class TabBarVC: UITabBarController {

    // 底部导航栏
    override func viewDidLoad() {
        super.viewDidLoad()
        // 设置背景颜色
        self.view.backgroundColor = UIColor.white
        
        // 基础
        let basicVC: BasicVC = BasicVC()
        let basicNavi: UINavigationController = UINavigationController(rootViewController: basicVC)
        
        // 进阶
        let advanceVC: AdvanceVC = AdvanceVC()
        let advanceNavi: UINavigationController = UINavigationController(rootViewController: advanceVC)
        
        // 高级
        let seniorVC: SeniorVC = SeniorVC()
        let seniorNavi: UINavigationController = UINavigationController(rootViewController: seniorVC)
        
        // 其他
        let otherVC: OtherVC = OtherVC()
        let otherNavi: UINavigationController = UINavigationController(rootViewController: otherVC)
        
        // 设置
        basicVC.tabBarItem.title = "基础"
        basicVC.title = basicVC.tabBarItem.title
        basicVC.tabBarItem.image = UIImage.init(named: "home_nor")
        basicVC.tabBarItem.selectedImage = UIImage.init(named: "home_sel")
        
        advanceVC.tabBarItem.title = "进阶"
        advanceVC.title = advanceVC.tabBarItem.title
        advanceVC.tabBarItem.image = UIImage.init(named: "mark_nor")
        advanceVC.tabBarItem.selectedImage = UIImage.init(named: "mark_sel")
        
        seniorVC.tabBarItem.title = "高级"
        seniorVC.title = seniorVC.tabBarItem.title
        seniorVC.tabBarItem.image = UIImage.init(named: "mine_nor")
        seniorVC.tabBarItem.selectedImage = UIImage.init(named: "mine_sel")
        
        otherVC.tabBarItem.title = "其他"
        otherVC.title = otherVC.tabBarItem.title
        otherVC.tabBarItem.image = UIImage.init(named: "setting_nor")
        otherVC.tabBarItem.selectedImage = UIImage.init(named: "setting_sel")
        
        self.setViewControllers([basicNavi, advanceNavi, seniorNavi, otherNavi], animated: false);
        
        // 设置徽标
        basicVC.navigationController?.tabBarItem.badgeValue = "1"
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
